components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx

```diff
@@ -296,124 +309,186 @@ export const ImportAccountModal = () => {
+            {hasImportError && (
+              <ErrorText>
+                {getLocale('braveWalletImportAccountError')}
+              </ErrorText>
+            )}
+
+            {importOption === 'key' ? (
               <Input
-                placeholder={`Origin ${getLocale(
-                  'braveWalletCreatePasswordInput'
-                )}`}
-                onChange={handlePasswordChanged}
+                placeholder={getLocale('braveWalletImportAccountPlaceholder')}
+                onBlur={clearClipboard}
                 type='password'
-                ref={passwordInputRef}
-              />
-            </>
-          )}
+                onInput={handlePrivateKeyChanged}
+                onKeyDown={handleKeyDown}
+              >
+                {
+                  // Label
+                  getLocale('braveWalletImportAccountKey')
+                }
+              </Input>
+            ) : (
+              <>
+                <ImportRow>
+                  <ImportButton htmlFor='recoverFile'>
+                    {getLocale('braveWalletImportAccountUploadButton')}
+                  </ImportButton>
+                  <DisclaimerText>
+                    {file
+                      ? reduceFileName(file[0].name)
+                      : getLocale('braveWalletImportAccountUploadPlaceholder')}
+                  </DisclaimerText>
+                </ImportRow>
+                <input
+                  type='file'
+                  id='recoverFile'
+                  name='recoverFile'
+                  style={{ display: 'none' }}
+                  onChange={onFileUpload}
+                  onClick={onClickFileUpload}
+                />
+                <Input
+                  placeholder={getLocale('braveWalletInputLabelPassword')}
+                  onInput={handlePasswordChanged}
+                  onKeyDown={handleKeyDown}
+                  onBlur={clearClipboard}
+                  type='password'
+                  ref={passwordInputRef}
+                >
+                  {
+                    // Label
+                    getLocale('braveWalletEnterPasswordIfApplicable')
+                  }
+                </Input>
+              </>
+            )}
 
-          <Input
-            value={accountName}
-            placeholder={getLocale('braveWalletAddAccountPlaceholder')}
-            onKeyDown={handleKeyDown}
-            onChange={handleAccountNameChanged}
-            autoFocus={true}
-          />
-
-          <NavButton
-            onSubmit={onClickCreateAccount}
-            disabled={isDisabled}
-            text={getLocale('braveWalletAddAccountImport')}
-            buttonType='primary'
-          />
+            <Input
+              value={accountName}
+              placeholder={getLocale('braveWalletAddAccountPlaceholder')}
+              onInput={handleAccountNameChanged}
+              onKeyDown={handleKeyDown}
+              showErrors={hasAccountNameError}
+              maxlength={30}
```

Comment:
BraveWallet.ACCOUNT_NAME_MAX_CHARACTER_LENGTH

Reply:
Thanks. I removed this prop value since the Nala component wasn't applying it
[a3e6c65](https://github.com/brave/brave-core/pull/22907/commits/a3e6c65516b717f2285997367093170880622356)